### ISSUE TITLE
[jvm-packages] - Tracker started, with env={} in Docker For Windows , Ubuntu Container -  with Spark 3.2.1

### ISSUE THREAD

**Issue by FelixDuong** [dmlc/xgboost#7700]
I met this error when running in Docker Ubuntu 20.04 with Spark 3.2.1... After reviewing some solutions to fix it via "xgboost-tracker.properties" and add "host-ip=172.17.0.2" , but it does not work... Anyone have  idea to solve it ..

```
import org.apache.spark.ml.{Pipeline, PipelineModel}
pipeline: org.apache.spark.ml.Pipeline = pipeline_8dcd16d2f790
res21: pipeline.type = pipeline_8dcd16d2f790
22/02/24 13:52:27 WARN XGBoostSpark: train_test_ratio is deprecated since XGBoost 0.82, we recommend to explicitly pass a training and multiple evaluation datasets by passing 'eval_sets' and 'eval_set_names'
Tracker started, with env={}
22/02/24 13:52:28 ERROR XGBoostSpark: the job was aborted due to
ml.dmlc.xgboost4j.java.XGBoostError: XGBoostModel training failed
        at ml.dmlc.xgboost4j.scala.spark.XGBoost$.postTrackerReturnProcessing(XGBoost.scala:750)
        at ml.dmlc.xgboost4j.scala.spark.XGBoost$.trainDistributed(XGBoost.scala:624)
        at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:196)
        at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:44)
        at org.apache.spark.ml.Predictor.fit(Predictor.scala:151)
        at org.apache.spark.ml.Predictor.fit(Predictor.scala:115)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$5(Pipeline.scala:151)
        at org.apache.spark.ml.MLEvents.withFitEvent(events.scala:130)
        at org.apache.spark.ml.MLEvents.withFitEvent$(events.scala:123)
        at org.apache.spark.ml.util.Instrumentation.withFitEvent(Instrumentation.scala:42)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$4(Pipeline.scala:151)
        at scala.collection.Iterator.foreach(Iterator.scala:943)
        at scala.collection.Iterator.foreach$(Iterator.scala:943)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$2(Pipeline.scala:147)
        at org.apache.spark.ml.MLEvents.withFitEvent(events.scala:130)
        at org.apache.spark.ml.MLEvents.withFitEvent$(events.scala:123)
        at org.apache.spark.ml.util.Instrumentation.withFitEvent(Instrumentation.scala:42)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$1(Pipeline.scala:133)
        at org.apache.spark.ml.util.Instrumentation$.$anonfun$instrumented$1(Instrumentation.scala:191)
        at scala.util.Try$.apply(Try.scala:213)
        at org.apache.spark.ml.util.Instrumentation$.instrumented(Instrumentation.scala:191)
        at org.apache.spark.ml.Pipeline.fit(Pipeline.scala:133)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:36)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:40)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:42)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:44)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:46)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:48)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:50)
        at $line67.$read$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:52)
        at $line67.$read$$iw$$iw$$iw.<init>(BehaviorCF.scala:54)
        at $line67.$read$$iw$$iw.<init>(BehaviorCF.scala:56)
        at $line67.$read$$iw.<init>(BehaviorCF.scala:58)
        at $line67.$read.<init>(BehaviorCF.scala:60)
        at $line67.$read$.<init>(BehaviorCF.scala:64)
        at $line67.$read$.<clinit>(BehaviorCF.scala)
        at $line67.$eval$.$print$lzycompute(BehaviorCF.scala:7)
        at $line67.$eval$.$print(BehaviorCF.scala:6)
        at $line67.$eval.$print(BehaviorCF.scala)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:747)
        at scala.tools.nsc.interpreter.IMain$Request.loadAndRun(IMain.scala:1020)
        at scala.tools.nsc.interpreter.IMain.$anonfun$interpret$1(IMain.scala:568)
        at scala.reflect.internal.util.ScalaClassLoader.asContext(ScalaClassLoader.scala:36)
        at scala.reflect.internal.util.ScalaClassLoader.asContext$(ScalaClassLoader.scala:116)
        at scala.reflect.internal.util.AbstractFileClassLoader.asContext(AbstractFileClassLoader.scala:41)
        at scala.tools.nsc.interpreter.IMain.loadAndRunReq$1(IMain.scala:567)
        at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:594)
        at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:564)
        at scala.tools.nsc.interpreter.ILoop.interpretStartingWith(ILoop.scala:865)
        at scala.tools.nsc.interpreter.ILoop.command(ILoop.scala:733)
        at scala.tools.nsc.interpreter.ILoop.processLine(ILoop.scala:435)
        at scala.tools.nsc.interpreter.ILoop.loop(ILoop.scala:456)
        at scala.tools.nsc.interpreter.ILoop.loop(ILoop.scala:451)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$interpretAllFrom$3(ILoop.scala:468)
        at scala.reflect.io.Streamable$Chars.applyReader(Streamable.scala:118)
        at scala.reflect.io.Streamable$Chars.applyReader$(Streamable.scala:116)
        at scala.reflect.io.File.applyReader(File.scala:50)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$interpretAllFrom$2(ILoop.scala:464)
        at scala.tools.nsc.interpreter.ILoop.savingReplayStack(ILoop.scala:97)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$interpretAllFrom$1(ILoop.scala:464)
        at scala.tools.nsc.interpreter.ILoop.savingReader(ILoop.scala:102)
        at scala.tools.nsc.interpreter.ILoop.interpretAllFrom(ILoop.scala:463)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$loadCommand$1(ILoop.scala:629)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$withFile$1(ILoop.scala:622)
        at scala.tools.nsc.interpreter.IMain.withLabel(IMain.scala:111)
        at scala.tools.nsc.interpreter.ILoop.withFile(ILoop.scala:621)
        at scala.tools.nsc.interpreter.ILoop.run$3(ILoop.scala:628)
        at scala.tools.nsc.interpreter.ILoop.loadCommand(ILoop.scala:635)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$standardCommands$9(ILoop.scala:194)
        at scala.tools.nsc.interpreter.LoopCommands$LineCmd.apply(LoopCommands.scala:161)
        at scala.tools.nsc.interpreter.LoopCommands.colonCommand(LoopCommands.scala:121)
        at scala.tools.nsc.interpreter.LoopCommands.colonCommand$(LoopCommands.scala:119)
        at scala.tools.nsc.interpreter.ILoop.colonCommand(ILoop.scala:48)
        at scala.tools.nsc.interpreter.ILoop.command(ILoop.scala:731)
        at scala.tools.nsc.interpreter.ILoop.processLine(ILoop.scala:435)
        at scala.tools.nsc.interpreter.ILoop.loop(ILoop.scala:456)
        at org.apache.spark.repl.SparkILoop.process(SparkILoop.scala:239)
        at org.apache.spark.repl.Main$.doMain(Main.scala:78)
        at org.apache.spark.repl.Main$.main(Main.scala:58)
        at org.apache.spark.repl.Main.main(Main.scala)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
        at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:955)
        at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
        at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
        at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
        at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1043)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1052)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
22/02/24 13:52:28 ERROR RabitTracker: Uncaught exception thrown by worker:
java.lang.IllegalStateException: SparkContext has been shutdown
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2206)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2235)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2254)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2279)
        at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$1(RDD.scala:1020)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
        at org.apache.spark.rdd.RDD.withScope(RDD.scala:414)
        at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:1018)
        at ml.dmlc.xgboost4j.scala.spark.XGBoost$$anon$1.run(XGBoost.scala:612)
22/02/24 13:52:28 ERROR RabitTracker: java.lang.InterruptedException: sleep interrupted
22/02/24 13:52:28 ERROR Instrumentation: ml.dmlc.xgboost4j.java.XGBoostError: XGBoostModel training failed
        at ml.dmlc.xgboost4j.scala.spark.XGBoost$.postTrackerReturnProcessing(XGBoost.scala:750)
        at ml.dmlc.xgboost4j.scala.spark.XGBoost$.trainDistributed(XGBoost.scala:624)
        at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:196)
        at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:44)
        at org.apache.spark.ml.Predictor.fit(Predictor.scala:151)
        at org.apache.spark.ml.Predictor.fit(Predictor.scala:115)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$5(Pipeline.scala:151)
        at org.apache.spark.ml.MLEvents.withFitEvent(events.scala:130)
        at org.apache.spark.ml.MLEvents.withFitEvent$(events.scala:123)
        at org.apache.spark.ml.util.Instrumentation.withFitEvent(Instrumentation.scala:42)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$4(Pipeline.scala:151)
        at scala.collection.Iterator.foreach(Iterator.scala:943)
        at scala.collection.Iterator.foreach$(Iterator.scala:943)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$2(Pipeline.scala:147)
        at org.apache.spark.ml.MLEvents.withFitEvent(events.scala:130)
        at org.apache.spark.ml.MLEvents.withFitEvent$(events.scala:123)
        at org.apache.spark.ml.util.Instrumentation.withFitEvent(Instrumentation.scala:42)
        at org.apache.spark.ml.Pipeline.$anonfun$fit$1(Pipeline.scala:133)
        at org.apache.spark.ml.util.Instrumentation$.$anonfun$instrumented$1(Instrumentation.scala:191)
        at scala.util.Try$.apply(Try.scala:213)
        at org.apache.spark.ml.util.Instrumentation$.instrumented(Instrumentation.scala:191)
        at org.apache.spark.ml.Pipeline.fit(Pipeline.scala:133)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:36)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:40)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:42)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:44)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:46)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:48)
        at $line67.$read$$iw$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:50)
        at $line67.$read$$iw$$iw$$iw$$iw.<init>(BehaviorCF.scala:52)
        at $line67.$read$$iw$$iw$$iw.<init>(BehaviorCF.scala:54)
        at $line67.$read$$iw$$iw.<init>(BehaviorCF.scala:56)
        at $line67.$read$$iw.<init>(BehaviorCF.scala:58)
        at $line67.$read.<init>(BehaviorCF.scala:60)
        at $line67.$read$.<init>(BehaviorCF.scala:64)
        at $line67.$read$.<clinit>(BehaviorCF.scala)
        at $line67.$eval$.$print$lzycompute(BehaviorCF.scala:7)
        at $line67.$eval$.$print(BehaviorCF.scala:6)
        at $line67.$eval.$print(BehaviorCF.scala)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:747)
        at scala.tools.nsc.interpreter.IMain$Request.loadAndRun(IMain.scala:1020)
        at scala.tools.nsc.interpreter.IMain.$anonfun$interpret$1(IMain.scala:568)
        at scala.reflect.internal.util.ScalaClassLoader.asContext(ScalaClassLoader.scala:36)
        at scala.reflect.internal.util.ScalaClassLoader.asContext$(ScalaClassLoader.scala:116)
        at scala.reflect.internal.util.AbstractFileClassLoader.asContext(AbstractFileClassLoader.scala:41)
        at scala.tools.nsc.interpreter.IMain.loadAndRunReq$1(IMain.scala:567)
        at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:594)
        at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:564)
        at scala.tools.nsc.interpreter.ILoop.interpretStartingWith(ILoop.scala:865)
        at scala.tools.nsc.interpreter.ILoop.command(ILoop.scala:733)
        at scala.tools.nsc.interpreter.ILoop.processLine(ILoop.scala:435)
        at scala.tools.nsc.interpreter.ILoop.loop(ILoop.scala:456)
        at scala.tools.nsc.interpreter.ILoop.loop(ILoop.scala:451)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$interpretAllFrom$3(ILoop.scala:468)
        at scala.reflect.io.Streamable$Chars.applyReader(Streamable.scala:118)
        at scala.reflect.io.Streamable$Chars.applyReader$(Streamable.scala:116)
        at scala.reflect.io.File.applyReader(File.scala:50)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$interpretAllFrom$2(ILoop.scala:464)
        at scala.tools.nsc.interpreter.ILoop.savingReplayStack(ILoop.scala:97)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$interpretAllFrom$1(ILoop.scala:464)
        at scala.tools.nsc.interpreter.ILoop.savingReader(ILoop.scala:102)
        at scala.tools.nsc.interpreter.ILoop.interpretAllFrom(ILoop.scala:463)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$loadCommand$1(ILoop.scala:629)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$withFile$1(ILoop.scala:622)
        at scala.tools.nsc.interpreter.IMain.withLabel(IMain.scala:111)
        at scala.tools.nsc.interpreter.ILoop.withFile(ILoop.scala:621)
        at scala.tools.nsc.interpreter.ILoop.run$3(ILoop.scala:628)
        at scala.tools.nsc.interpreter.ILoop.loadCommand(ILoop.scala:635)
        at scala.tools.nsc.interpreter.ILoop.$anonfun$standardCommands$9(ILoop.scala:194)
        at scala.tools.nsc.interpreter.LoopCommands$LineCmd.apply(LoopCommands.scala:161)
        at scala.tools.nsc.interpreter.LoopCommands.colonCommand(LoopCommands.scala:121)
        at scala.tools.nsc.interpreter.LoopCommands.colonCommand$(LoopCommands.scala:119)
        at scala.tools.nsc.interpreter.ILoop.colonCommand(ILoop.scala:48)
        at scala.tools.nsc.interpreter.ILoop.command(ILoop.scala:731)
        at scala.tools.nsc.interpreter.ILoop.processLine(ILoop.scala:435)
        at scala.tools.nsc.interpreter.ILoop.loop(ILoop.scala:456)
        at org.apache.spark.repl.SparkILoop.process(SparkILoop.scala:239)
        at org.apache.spark.repl.Main$.doMain(Main.scala:78)
        at org.apache.spark.repl.Main$.main(Main.scala:58)
        at org.apache.spark.repl.Main.main(Main.scala)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
        at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:955)
        at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
        at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
        at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
        at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1043)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1052)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)

ml.dmlc.xgboost4j.java.XGBoostError: XGBoostModel training failed
  at ml.dmlc.xgboost4j.scala.spark.XGBoost$.postTrackerReturnProcessing(XGBoost.scala:750)
  at ml.dmlc.xgboost4j.scala.spark.XGBoost$.trainDistributed(XGBoost.scala:624)
  at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:196)
  at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:44)
  at org.apache.spark.ml.Predictor.fit(Predictor.scala:151)
  at org.apache.spark.ml.Predictor.fit(Predictor.scala:115)
  at org.apache.spark.ml.Pipeline.$anonfun$fit$5(Pipeline.scala:151)
  at org.apache.spark.ml.MLEvents.withFitEvent(events.scala:130)
  at org.apache.spark.ml.MLEvents.withFitEvent$(events.scala:123)
  at org.apache.spark.ml.util.Instrumentation.withFitEvent(Instrumentation.scala:42)
  at org.apache.spark.ml.Pipeline.$anonfun$fit$4(Pipeline.scala:151)
  at scala.collection.Iterator.foreach(Iterator.scala:943)
  at scala.collection.Iterator.foreach$(Iterator.scala:943)
  at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
  at org.apache.spark.ml.Pipeline.$anonfun$fit$2(Pipeline.scala:147)
  at org.apache.spark.ml.MLEvents.withFitEvent(events.scala:130)
  at org.apache.spark.ml.MLEvents.withFitEvent$(events.scala:123)
  at org.apache.spark.ml.util.Instrumentation.withFitEvent(Instrumentation.scala:42)
  at org.apache.spark.ml.Pipeline.$anonfun$fit$1(Pipeline.scala:133)
  at org.apache.spark.ml.util.Instrumentation$.$anonfun$instrumented$1(Instrumentation.scala:191)
  at scala.util.Try$.apply(Try.scala:213)
  at org.apache.spark.ml.util.Instrumentation$.instrumented(Instrumentation.scala:191)
  at org.apache.spark.ml.Pipeline.fit(Pipeline.scala:133)
  ... 74 elided```

**Comment by lcx517** [comment]
I met the same problem as "Tracker started, with env={}", and hung on the same code line.
I ran with Spark 3.1.1, and dependencies as below:
        <dependency>
            <groupId>ml.dmlc</groupId>
            <artifactId>xgboost4j_2.12</artifactId>
            <version>1.5.2</version>
        </dependency>
        <dependency>
            <groupId>ml.dmlc</groupId>
            <artifactId>xgboost4j-spark_2.12</artifactId>
            <version>1.5.2</version>
        </dependency>

**Comment by trivialfis** [comment]
We recently added a user option in dask for others to specify the host IP and port.  I'm not entirely sure how to replicate that in spark package.

**Comment by FelixDuong** [comment]
> We recently added a user option in dask for others to specify the host IP and port. I'm not entirely sure how to replicate that in spark package.

Thanks trivialfis, I am looking forward to trying in new next version . !!!

**Comment by lcx517** [comment]
> We recently added a user option in dask for others to specify the host IP and port. I'm not entirely sure how to replicate that in spark package.

Thank you. I can see env values (such as DMLC_TRACKER_URI, DMLC_TRACKER_PORT...etc) correctly when I ran locally. However, when I submit to spark cluster, the env is empty. It will be set automatically in spark 2.x, and I'm not sure it's a good idea to specify the dynamic cluster IP or port manually...

**Comment by trivialfis** [comment]
> I'm not sure it's a good idea to specify the dynamic cluster IP or port manually

For the Python dask module, it's the last resort.  Normally XGBoost can obtain the IP and port automatically. However, sometimes the host might be incorrectly configured in clusters, or sometimes there are network configurations that we don't understand, or sometimes it's simply caused by docker restricting available ports.  In these cases, letting the users to specify the IP and port might be the only possible way to get things working.

**Comment by trivialfis** [comment]
> Thanks trivialfis, I am looking forward to trying in new next version . !!!

Thank you, meanwhile, if you have suggestions on how to port that to spark package in a natural way please share.

**Comment by FelixDuong** [comment]
Indeep, I have tested xgboost4j-spark in some environments and realized couple ideas:
- With xgboost4j-spark 1.5.2 and run in real Ubuntu (not virtually) and run in VMware (Windows hosts) , it works well!!! Smoothly!!
- With xgboost4j-spark 1.5.2 and run in docker for Windows or for Mac , it encounters error env={}, maybe it can not capture ip's docker container. In new docker versions, there is no docker0 in windows and mac! 
- Latest version Spark still run smoothly on docker for Windows and Mac with GBT (Gradient Boost Tree model)--> So, in my opinion, the problem we encounter just internal ip in docker or running evn which xgboost4j-spark can not capture properly .
    Sorry, I don't have much experience in porting to spark , all there just my logic from errors encountered. Hopefully it can help you. I am looking forward receiving good news from our community- Xgboost.
     Regards,

**Comment by wbo4958** [comment]
@FelixDuong I just had a PR to fix this issue by specifying the rabit tracker host IP
``` scala
val hostIP = xx.xx.xx.xx
"tracker_conf" -> TrackerConf(0L, "python", hostIp)
```

The PR will be shipped in 1.6.0 release.
